<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content='A differential abundance method for the analysis of microbiome data. radEmu estimates fold-differences in the abundance of taxa across samples relative to "typical" fold-differences. Notably, it does not require pseudocounts, nor choosing a denominator taxon. '>
<title>Using Relative Abundance Data to Estimate of Multiplicative Differences in Mean Absolute Abundance  • radEmu</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Using Relative Abundance Data to Estimate of Multiplicative Differences in Mean Absolute Abundance ">
<meta property="og:description" content='A differential abundance method for the analysis of microbiome data. radEmu estimates fold-differences in the abundance of taxa across samples relative to "typical" fold-differences. Notably, it does not require pseudocounts, nor choosing a denominator taxon. '>
<meta property="og:image" content="https://statdivlab.github.io/radEmu/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">radEmu</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/intro_radEmu.html">Introduction to radEmu</a>
    <a class="dropdown-item" href="articles/intro_radEmu_with_phyloseq.html">Introduction to radEmu with phyloseq</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/statdivlab/radEmu/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="rademu-">radEmu <a class="anchor" aria-label="anchor" href="#rademu-"></a>
</h1>
</div>
<!-- badges: start -->

<p><code>radEmu</code> is an <code>R</code> package for estimating changes in the abundance of microbial taxa using amplicon or shotgun sequencing technologies.</p>
<p>If you are a <strong>microbial ecologist</strong> or <strong>bioinformatician</strong>, some of the things that you may like about <code>radEmu</code> include</p>
<ul>
<li>
<code>radEmu</code> uses your amplicon or shotgun sequencing to estimate changes in the “absolute abundance” of microbial taxa. Here, “absolute abundance” could be interpreted on the cell count, cell concentration or DNA concentration scale. Yes! It’s true!
<ul>
<li>We know this sounds magical! You can check out Section 2 of the manuscript for details.</li>
<li>In brief, we <em>can’t</em> recover the absolute abundance of taxa in <em>any individual sample</em> from amplicon or shotgun sequencing. However, we <em>can</em> estimate fold-differences in abundances <em>across samples</em>.</li>
</ul>
</li>
<li>
<code>radEmu</code> formalizes some of the nice things about log-ratio-type methods for differential abundance, including
<ul>
<li>
<code>radEmu</code> is robust to differential detection of taxa, so you don’t have to worry about (e.g.) the different extraction/PCR efficiency of your protocol</li>
<li>
<code>radEmu</code> is robust to unequal sampling effort. No need to rarefy! (Actually, please <a href="https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2019.02407/full" class="external-link">don’t</a>.)</li>
<li>
<code>radEmu</code> deals with zeroes natively, without any need for arbitrary parameters like <em>pseudocounts</em>
</li>
<li>
<code>radEmu</code> does not require that you have a “reference taxon” that is not changing in abundance across samples
<ul>
<li>Instead, <code>radEmu</code> estimates differences in abundance across taxa</li>
</ul>
</li>
</ul>
</li>
<li>Amongst existing methods, <code>radEmu</code> is <em>most similar in flavor</em> to ALDEx2 and ANCOM (and ANCOM relatives), but doesn’t require priors, log-ratio transformations (and thus pseudocounts), nor a reference taxon!</li>
<li>
<code>radEmu</code> can adjust for relevant covariates, including precision variables and confounders</li>
<li>
<code>radEmu</code> achieves all of the above by jointly modeling all taxa (i.e., it’s not a taxon-by-taxon model like corncob). This makes it harder to parallelize, but fortunately <em>testing</em> can be parallelized easily. (There’s is an example in the preprint’s <a href="https://github.com/statdivlab/radEmu_supplementary" class="external-link">supplementary material</a>, but let us know if you want a tutorial on <em>how</em>!) On a standard desktop, <code>radEmu</code> can handle 1000 taxa, 800 samples and 12 covariates. You may want to get a 35-minute coffee break while it runs, though.</li>
<li>
<code>radEmu</code> is publicly available in open-source software… right here!</li>
</ul>
<p>If you are a <strong>statistician</strong>, some of the things that you may like about <code>radEmu</code> include</p>
<ul>
<li>A clearly defined and interpretable target estimand</li>
<li>Fast algorithms for estimation under alternative and null hypotheses</li>
<li>Type 1 error rate control even under pathological distribution misspecification <em>and</em> small sample sizes… at the same time!
<ul>
<li>Check out the <a href="https://arxiv.org/abs/2402.05231" class="external-link">preprint</a> for details! Our model isn’t built to cater to zero-inflated Negative Binominally distributed data, but it still did awesome!</li>
<li>Note that the robust score tests have better error rate control than the robust Wald tests (they are a bit slower)</li>
</ul>
</li>
</ul>
<p>Sadly we do not yet have a <del>logo</del> nice-looking logo. If you would like to design us one, please let Amy know!</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To download the radEmu package, use the code below.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"statdivlab/radEmu"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statdivlab/radEmu" class="external-link">radEmu</a></span><span class="op">)</span></span></code></pre></div>
<p>We are currently only releasing <code>radEmu</code> via GitHub. If you’d like us to consider submitting to CRAN, please let us know by opening an issue.</p>
</div>
<div class="section level2">
<h2 id="use">Use<a class="anchor" aria-label="anchor" href="#use"></a>
</h2>
<p>The vignette demonstrates example usage of the main functions. Please <a href="https://github.com/statdivlab/radEmu/issues" class="external-link">file an issue</a> if you have a request for a tutorial that is not currently included. The following code shows the easy-to-use syntax if your data is in a <code>phyloseq</code> object:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span> <span class="op">+</span> <span class="va">Study</span> <span class="op">+</span> <span class="va">Gender</span> <span class="op">+</span> <span class="va">Sampling</span>, </span>
<span>                 Y <span class="op">=</span> <span class="va">my_phyloseq_object</span><span class="op">)</span> </span></code></pre></div>
<p>and if your abundances and covariates are in a dataframe, you can use the following:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span> <span class="op">+</span> <span class="va">Study</span> <span class="op">+</span> <span class="va">Gender</span> <span class="op">+</span> <span class="va">Sampling</span>,</span>
<span>                  data <span class="op">=</span> <span class="va">my_covariates_df</span>, </span>
<span>                  Y <span class="op">=</span> <span class="va">my_abundances_df</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use <code>radEmu</code> for your analysis, please cite our open-access preprint, available on arXiv.</p>
<p>David S Clausen and Amy D Willis. 2024+. “Estimating Fold Changes from Partially Observed Outcomes with Applications in Microbial Metagenomics.” <a href="https://arxiv.org/abs/2402.05231" class="external-link">arxiv.org/abs/2402.05231</a></p>
<p>Huge thanks to the NIGMS for funding this work through Amy’s R35!</p>
</div>
<div class="section level2">
<h2 id="bug-reports--change-requests">Bug Reports / Change Requests<a class="anchor" aria-label="anchor" href="#bug-reports--change-requests"></a>
</h2>
<p>If you encounter a bug or would like make a change request, please file it as an issue <a href="https://github.com/statdivlab/radEmu/issues" class="external-link">here</a>.</p>
<p>If you’re a developer, we would love to review your pull requests.</p>
</div>
<div class="section level2">
<h2 id="nomenclature">Nomenclature<a class="anchor" aria-label="anchor" href="#nomenclature"></a>
</h2>
<p>When we are not developing fast, robust and interpretable estimation methods, we enjoy making up silly names for our fast, robust and interpretable estimation methods. <code>radEmu</code> abbreviates <code>radEmuAbPill</code>, which denotes “using <strong>r</strong>elative <strong>a</strong>bundance <strong>d</strong>ata to <strong>e</strong>stimate <strong>mu</strong>ltiplicative differences in <strong>ab</strong>solute abundances with <strong>p</strong>artially <strong>i</strong>dentified <strong>l</strong>og-<strong>l</strong>inear models.”</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/statdivlab/radEmu/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing radEmu</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>David Clausen <br><small class="roles"> Author, maintainer </small>  </li>
<li>Amy Willis <br><small class="roles"> Author </small>  </li>
<li>Sarah Teichman <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/statdivlab/radEmu/actions" class="external-link"><img src="https://github.com/statdivlab/radEmu/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/github/statdivlab/radEmu" class="external-link"><img src="https://codecov.io/github/statdivlab/radEmu/coverage.svg?branch=main" alt="codecov"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by David Clausen, Amy Willis, Sarah Teichman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
