<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to radEmu with TreeSummarizedExperiment • radEmu</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to radEmu with TreeSummarizedExperiment">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">radEmu</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/intro_radEmu_with_phyloseq.html">Introduction to radEmu with phyloseq</a></li>
    <li><a class="dropdown-item" href="../articles/intro_radEmu_with_tse.html">Introduction to radEmu with TreeSummarizedExperiment</a></li>
    <li><a class="dropdown-item" href="../articles/intro_radEmu.html">Introduction to radEmu</a></li>
    <li><a class="dropdown-item" href="../articles/parallel_radEmu.html">Parallelizing computation for score tests with radEmu</a></li>
    <li><a class="dropdown-item" href="../articles/radEmu_clustered_data.html">Using radEmu on clustered data</a></li>
    <li><a class="dropdown-item" href="../articles/radEmu_with_reference_taxon.html">Using radEmu with a reference taxon</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/statdivlab/radEmu/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to radEmu with TreeSummarizedExperiment</h1>
                        <h4 data-toc-skip class="author">David Clausen,
Sarah Teichman and Amy Willis</h4>
            
            <h4 data-toc-skip class="date">2025-03-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statdivlab/radEmu/blob/main/vignettes/intro_radEmu_with_tse.Rmd" class="external-link"><code>vignettes/intro_radEmu_with_tse.Rmd</code></a></small>
      <div class="d-none name"><code>intro_radEmu_with_tse.Rmd</code></div>
    </div>

    
    
<p>First, we will install <code>radEmu</code>, if we haven’t
already.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!require("remotes", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("remotes")</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># remotes::install_github("statdivlab/radEmu")</span></span></code></pre></div>
<p>Next, we can load <code>radEmu</code> as well as the
<code>tidyverse</code> package suite.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statdivlab/radEmu" class="external-link">radEmu</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette provides an introduction to using <code>radEmu</code>
for differential abundance analysis using a
<code>TreeSummarizedExperiment</code> data object. For more in-depth
explanations of how this software works and details on this analysis,
see the vignette “intro_radEmu.Rmd”.</p>
<p>In this lab we’ll explore a <a href="https://www.nature.com/articles/s41591-019-0406-6" class="external-link">dataset
published by Wirbel et al. (2019)</a>. This is a meta-analysis of
case-control studies, meaning that Wirbel et al. collected raw
sequencing data from studies other researchers conducted and re-analyzed
it (in this case, they also collected some new data of their own).</p>
<p>Wirbel et al. published two pieces of data we’ll focus on today:</p>
<ul>
<li>metadata giving demographics and other information about
participants</li>
<li>a mOTU (metagenomic OTU) table</li>
</ul>
<p>In the manuscript, we looked at differential abundance across
otherwise similar colorectal cancer and non-cancer control study
participants for the 849 mOTUs that Wirbel et al. published. For the
purpose of having a streamlined tutorial, we will only look at a subset
of those 849 mOTUs in this vignette.</p>
</div>
<div class="section level2">
<h2 id="loading-and-exploring-data">Loading and exploring data<a class="anchor" aria-label="anchor" href="#loading-and-exploring-data"></a>
</h2>
<p>Note that in order to follow along with this tutorial (but not to use
<code>radEmu</code>!) you will need to have
<code>TreeSummarizedExperiment</code> installed. We will check if you
have <code>TreeSummarizedExperiment</code> installed, and if you do not
then you can read the following code but it will not be run.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"TreeSummarizedExperiment"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">==</span> <span class="cn">TRUE</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "TreeSummarizedExperiment is installed: TRUE"</span></span></code></pre>
<p>Now that we have loaded the <code>TreeSummarizedExperiment</code>
package, we will create our <code>TreeSummarizedExperiment</code> data
object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TreeSummarizedExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wirbel_otu</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wirbel_taxonomy</span><span class="op">)</span></span>
<span><span class="va">wirbel_tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-constructor.html" class="external-link">TreeSummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">wirbel_otu</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                       rowData <span class="op">=</span> <span class="va">wirbel_taxonomy</span>,</span>
<span>                                       colData <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">)</span></span>
<span><span class="va">wirbel_tse</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 845 566 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): Count</span></span>
<span><span class="co">#&gt; rownames(845): Streptococcus anginosus [ref_mOTU_v2_0004]</span></span>
<span><span class="co">#&gt;   Enterobacteriaceae sp. [ref_mOTU_v2_0036] ... unknown Clostridiales</span></span>
<span><span class="co">#&gt;   [meta_mOTU_v2_7795] unknown Clostridiales [meta_mOTU_v2_7800]</span></span>
<span><span class="co">#&gt; rowData names(7): domain phylum ... genus species</span></span>
<span><span class="co">#&gt; colnames(566): CCIS00146684ST.4.0 CCIS00281083ST.3.0 ... SAMEA3178940</span></span>
<span><span class="co">#&gt;   SAMEA3178943</span></span>
<span><span class="co">#&gt; colData names(14): Sample_ID External_ID ... BMI_spline.2 Sampling</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: NULL</span></span>
<span><span class="co">#&gt; rowTree: NULL</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre></div>
<p>We’ll start by looking at the metadata.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">wirbel_tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 566  14</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">wirbel_tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 14 columns</span></span>
<span><span class="co">#&gt;                             Sample_ID External_ID       Age      Gender</span></span>
<span><span class="co">#&gt;                           &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0 CCIS00146684ST-4-0      FR-726        72           F</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0 CCIS00281083ST-3-0      FR-060        53           M</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0 CCIS02124300ST-4-0      FR-568        35           M</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0 CCIS02379307ST-4-0      FR-828        67           M</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0 CCIS02856720ST-4-0      FR-027        74           M</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0 CCIS03473770ST-4-0      FR-192        29           M</span></span>
<span><span class="co">#&gt;                          BMI     Country       Study       Group Library_Size</span></span>
<span><span class="co">#&gt;                    &lt;numeric&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;integer&gt;</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0        25         FRA      FR-CRC         CTR     35443944</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0        32         FRA      FR-CRC         CTR     19307896</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0        23         FRA      FR-CRC         CTR     42141246</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0        28         FRA      FR-CRC         CRC      4829533</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0        27         FRA      FR-CRC         CTR     34294675</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0        24         FRA      FR-CRC         CTR     20262319</span></span>
<span><span class="co">#&gt;                    Age_spline.1 Age_spline.2 BMI_spline.1 BMI_spline.2</span></span>
<span><span class="co">#&gt;                       &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0   -0.1975543     0.738962    1.1898242    -0.560692</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0   -0.0812613    -0.681853   -1.4067931     2.003914</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0   -2.1745353    -0.681853    0.4547668    -0.670604</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0    0.6746432    -0.149048    0.0769882     0.538425</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0   -0.5464333     1.094166    0.4479336     0.172053</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0   -2.8722933    -0.681853    0.9526144    -0.670604</span></span>
<span><span class="co">#&gt;                       Sampling</span></span>
<span><span class="co">#&gt;                    &lt;character&gt;</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0      BEFORE</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0      BEFORE</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0      BEFORE</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0      BEFORE</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0      BEFORE</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0      BEFORE</span></span></code></pre></div>
<p>We can see that this dataset includes
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>566</mn><annotation encoding="application/x-tex">566</annotation></semantics></math>
observations and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>14</mn><annotation encoding="application/x-tex">14</annotation></semantics></math>
variables.</p>
<p>Now let’s look at the mOTU table.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_tse</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 845 566</span></span>
<span><span class="co"># let's check out a subset</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_tse</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                                     CCIS00146684ST.4.0</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]                           0</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]                            3</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                                   0</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079]                  0</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]                             0</span></span>
<span><span class="co">#&gt;                                                     CCIS00281083ST.3.0</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]                           0</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]                            0</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                                   0</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079]                  0</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]                             0</span></span>
<span><span class="co">#&gt;                                                     CCIS02124300ST.4.0</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]                           2</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]                            5</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                                   0</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079]                  0</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]                             6</span></span></code></pre></div>
<p>We can see that this table has
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>566</mn><annotation encoding="application/x-tex">566</annotation></semantics></math>
samples (just like the metadata) and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>845</mn><annotation encoding="application/x-tex">845</annotation></semantics></math>
mOTUs. Let’s save these mOTU names in a vector.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mOTU_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_tse</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Finally, we can check out the taxonomy table.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">wirbel_tse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 7 columns</span></span>
<span><span class="co">#&gt;                                                          domain         phylum</span></span>
<span><span class="co">#&gt;                                                     &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]             Bacteria      Bacillota</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]              Bacteria Pseudomonadota</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                     Bacteria Pseudomonadota</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079]    Bacteria Pseudomonadota</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]               Bacteria      Bacillota</span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]            Bacteria      Bacillota</span></span>
<span><span class="co">#&gt;                                                                   class</span></span>
<span><span class="co">#&gt;                                                             &lt;character&gt;</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]                      Bacilli</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           Gammaproteobacteria</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  Gammaproteobacteria</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] Gammaproteobacteria</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]                        Bacilli</span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]                     Bacilli</span></span>
<span><span class="co">#&gt;                                                                order</span></span>
<span><span class="co">#&gt;                                                          &lt;character&gt;</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]           Lactobacillales</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           Enterobacterales</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  Enterobacterales</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] Enterobacterales</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]             Lactobacillales</span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]          Lactobacillales</span></span>
<span><span class="co">#&gt;                                                                 family</span></span>
<span><span class="co">#&gt;                                                            &lt;character&gt;</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]            Streptococcaceae</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           Enterobacteriaceae</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  Enterobacteriaceae</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] Enterobacteriaceae</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]               Enterococcaceae</span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]           Lactobacillaceae</span></span>
<span><span class="co">#&gt;                                                                  genus</span></span>
<span><span class="co">#&gt;                                                            &lt;character&gt;</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]               Streptococcus</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           Enterobacteriaceae</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                         Citrobacter</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079]         Klebsiella</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]                  Enterococcus</span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]              Lactobacillus</span></span>
<span><span class="co">#&gt;                                                                    species</span></span>
<span><span class="co">#&gt;                                                                &lt;character&gt;</span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          Streptococcus angino..</span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           Enterobacteriaceae b..</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                         Citrobacter sp.</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] Klebsiella michigane..</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]             Enterococcus faecalis</span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         Ligilactobacillus sa..</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fitting-a-model">Fitting a model<a class="anchor" aria-label="anchor" href="#fitting-a-model"></a>
</h2>
<p><code>radEmu</code> is a package that can be used to estimate
fold-differences in the abundance of microbial taxa between levels of a
covariate. In this analysis, the covariate that we are primarily
interested in is whether a sample is from a case of colorectal cancer or
a control. We will make control (“CTR”) the reference category:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">wirbel_tse</span><span class="op">)</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">wirbel_tse</span><span class="op">)</span><span class="op">$</span><span class="va">Group</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CTR"</span>,<span class="st">"CRC"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>While in general we would fit a model to all mOTUs, we are going to
subset to some specific genera for the purposes of this tutorial. Let’s
look at <em>Eubacterium</em>, <em>Porphyromonas</em>,
<em>Faecalibacteria</em>, and <em>Fusobacterium</em> for now.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chosen_genera</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Eubacterium"</span>, <span class="st">"Faecalibacterium"</span>, <span class="st">"Fusobacterium"</span>, <span class="st">"Porphyromonas"</span><span class="op">)</span></span>
<span><span class="va">genera_rows</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">wirbel_tse</span><span class="op">)</span><span class="op">$</span><span class="va">genus</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">chosen_genera</span></span>
<span><span class="va">wirbel_restrict</span> <span class="op">&lt;-</span> <span class="va">wirbel_tse</span><span class="op">[</span><span class="va">genera_rows</span><span class="op">]</span></span></code></pre></div>
<p>Again, while we would generally fit a model using all of our samples,
for this tutorial we are only going to consider data from a case-control
study from China.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">wirbel_restrict</span><span class="op">)</span><span class="op">$</span><span class="va">Country</span> <span class="op">==</span> <span class="st">"CHI"</span></span>
<span><span class="va">wirbel_china</span> <span class="op">&lt;-</span> <span class="va">wirbel_restrict</span><span class="op">[</span>, <span class="va">sample_cols</span><span class="op">]</span></span></code></pre></div>
<p>Next, we want to confirm that all samples have at least one non-zero
count across the categories we’ve chosen and that all categories have at
least one non-zero count across the samples we’ve chosen.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_china</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># no samples have a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_china</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># one category has a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="va">category_to_rm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_china</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span></span>
<span><span class="va">wirbel_china</span> <span class="op">&lt;-</span> <span class="va">wirbel_china</span><span class="op">[</span><span class="op">!</span><span class="va">category_to_rm</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">wirbel_china</span>, <span class="st">"Count"</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># now no categories have a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<p>The function that we use to fit our model is called
<code>emuFit</code>. It can accept your data in various forms, and here
we will show how to use it with a <code>TreeSummarizedExperiment</code>
object as input.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span>, </span>
<span>                 Y <span class="op">=</span> <span class="va">wirbel_china</span>, </span>
<span>                 assay_name <span class="op">=</span> <span class="st">"Count"</span>,</span>
<span>                 run_score_tests <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></code></pre></div>
<p>The way to access estimated coefficients and confidence intervals
from the model is with <code>ch_fit$coef</code>.</p>
<p>Now, we can easily visualize our results using the
<code>plot.emuFit</code> function!</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ch_fit</span><span class="op">)</span><span class="op">$</span><span class="va">plots</span></span>
<span><span class="co">#&gt; $p1</span></span></code></pre></div>
<p><img src="intro_radEmu_with_tse_files/figure-html/unnamed-chunk-16-1.png" width="576"></p>
<p>If you’d like to see more explanations of the <code>radEmu</code>
software and additional analyses of this data, check out the vignette
“intro_radEmu.Rmd”.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Clausen, Amy D Willis, Sarah Teichman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
