<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to radEmu • radEmu</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to radEmu">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">radEmu</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/intro_radEmu_with_phyloseq.html">Introduction to radEmu with phyloseq</a></li>
    <li><a class="dropdown-item" href="../articles/intro_radEmu_with_tse.html">Introduction to radEmu with TreeSummarizedExperiment</a></li>
    <li><a class="dropdown-item" href="../articles/intro_radEmu.html">Introduction to radEmu</a></li>
    <li><a class="dropdown-item" href="../articles/parallel_radEmu.html">Parallelizing computation for score tests with radEmu</a></li>
    <li><a class="dropdown-item" href="../articles/radEmu_clustered_data.html">Using radEmu on clustered data</a></li>
    <li><a class="dropdown-item" href="../articles/radEmu_with_reference_taxon.html">Using radEmu with a reference taxon</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/statdivlab/radEmu/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to radEmu</h1>
                        <h4 data-toc-skip class="author">David Clausen,
Sarah Teichman and Amy Willis</h4>
            
            <h4 data-toc-skip class="date">2025-03-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statdivlab/radEmu/blob/main/vignettes/intro_radEmu.Rmd" class="external-link"><code>vignettes/intro_radEmu.Rmd</code></a></small>
      <div class="d-none name"><code>intro_radEmu.Rmd</code></div>
    </div>

    
    
<p>First, we will install <code>radEmu</code>, if we haven’t
already.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!require("remotes", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("remotes")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># remotes::install_github("statdivlab/radEmu")</span></span></code></pre></div>
<p>Next, we can load <code>radEmu</code> as well as the
<code>tidyverse</code> package suite.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statdivlab/radEmu" class="external-link">radEmu</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this lab we’ll explore a <a href="https://www.nature.com/articles/s41591-019-0406-6" class="external-link">dataset
published by Wirbel et al. (2019)</a>. This is a meta-analysis of
case-control studies, meaning that Wirbel et al. collected raw
sequencing data from studies other researchers conducted and re-analyzed
it (in this case, they also collected some new data of their own).</p>
<p>Wirbel et al. published two pieces of data we’ll focus on today:</p>
<ul>
<li>metadata giving demographics and other information about
participants</li>
<li>a mOTU (metagenomic OTU) table</li>
</ul>
<p>In the manuscript, we looked at differential abundance across
otherwise similar colorectal cancer and non-cancer control study
participants for the 849 mOTUs that Wirbel et al. published. For the
purpose of having a streamlined tutorial, we will only look at a subset
of those 849 mOTUs in this vignette.</p>
</div>
<div class="section level2">
<h2 id="loading-and-exploring-data">Loading and exploring data<a class="anchor" aria-label="anchor" href="#loading-and-exploring-data"></a>
</h2>
<p>We’ll start by looking at the metadata.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"wirbel_sample"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 566  14</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             Sample_ID External_ID Age Gender BMI Country  Study</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0 CCIS00146684ST-4-0      FR-726  72      F  25     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0 CCIS00281083ST-3-0      FR-060  53      M  32     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0 CCIS02124300ST-4-0      FR-568  35      M  23     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0 CCIS02379307ST-4-0      FR-828  67      M  28     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0 CCIS02856720ST-4-0      FR-027  74      M  27     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0 CCIS03473770ST-4-0      FR-192  29      M  24     FRA FR-CRC</span></span>
<span><span class="co">#&gt;                    Group Library_Size Age_spline.1 Age_spline.2 BMI_spline.1</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0   CTR     35443944  -0.19755428    0.7389621   1.18982420</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0   CTR     19307896  -0.08126128   -0.6818534  -1.40679307</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0   CTR     42141246  -2.17453529   -0.6818534   0.45476676</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0   CRC      4829533   0.67464323   -0.1490476   0.07698823</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0   CTR     34294675  -0.54643328    1.0941660   0.44793355</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0   CTR     20262319  -2.87229329   -0.6818534   0.95261443</span></span>
<span><span class="co">#&gt;                    BMI_spline.2 Sampling</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0   -0.5606919   BEFORE</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0    2.0039136   BEFORE</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0   -0.6706035   BEFORE</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0    0.5384247   BEFORE</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0    0.1720525   BEFORE</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0   -0.6706035   BEFORE</span></span></code></pre></div>
<p>We can see that this dataset includes
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>566</mn><annotation encoding="application/x-tex">566</annotation></semantics></math>
observations and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>14</mn><annotation encoding="application/x-tex">14</annotation></semantics></math>
variables. Let’s see how many observations we have among cases (“CRC”)
and controls (“CTR”)</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wirbel_sample</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Group</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">#&gt;   Group count</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CRC     278</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CTR     288</span></span></code></pre></div>
<p>We have data from studies in 5 different countries. How much from
each study, you ask? Let’s find out!</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wirbel_sample</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Country</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 2</span></span></span>
<span><span class="co">#&gt;   Country count</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> AUS       109</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CHI       126</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> FRA       110</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> GER       120</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> USA       101</span></span></code></pre></div>
<p>Let’s see how many cases and controls were enrolled in each study as
well.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wirbel_sample</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Country</span>, <span class="va">Group</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; `summarise()` has grouped output by 'Country'. You can override using the</span></span>
<span><span class="co">#&gt; `.groups` argument.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   Country [5]</span></span></span>
<span><span class="co">#&gt;    Country Group     n</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> AUS     CRC      46</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> AUS     CTR      63</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CHI     CRC      72</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CHI     CTR      54</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> FRA     CRC      51</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> FRA     CTR      59</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> GER     CRC      60</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GER     CTR      60</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> USA     CRC      49</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> USA     CTR      52</span></span></code></pre></div>
<p>Now let’s load the mOTU table.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"wirbel_otu"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">wirbel_otu</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 566 845</span></span>
<span><span class="co"># let's check out a subset</span></span>
<span><span class="va">wirbel_otu</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                    Streptococcus anginosus [ref_mOTU_v2_0004]</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0                                          0</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0                                          0</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0                                          2</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0                                          0</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0                                          1</span></span>
<span><span class="co">#&gt;                    Enterobacteriaceae sp. [ref_mOTU_v2_0036]</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0                                         3</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0                                         0</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0                                         5</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0                                         5</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0                                       675</span></span>
<span><span class="co">#&gt;                    Citrobacter sp. [ref_mOTU_v2_0076]</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0                                  0</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0                                  0</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0                                  0</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0                                  0</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0                                  0</span></span></code></pre></div>
<p>We can see that this table has
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>566</mn><annotation encoding="application/x-tex">566</annotation></semantics></math>
samples (just like the metadata) and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>845</mn><annotation encoding="application/x-tex">845</annotation></semantics></math>
mOTUs. Let’s save these mOTU names in a vector.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mOTU_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">wirbel_otu</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fitting-a-model">Fitting a model<a class="anchor" aria-label="anchor" href="#fitting-a-model"></a>
</h2>
<p><code>radEmu</code> is a package that can be used to estimate
fold-differences in the abundance of microbial taxa between levels of a
covariate. In this analysis, the covariate that we are primarily
interested in is whether a sample is from a case of colorectal cancer or
a control. We will make control (“CTR”) the reference category:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Group</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CTR"</span>,<span class="st">"CRC"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>While in general we would fit a model to all mOTUs, we are going to
subset to some specific genera for the purposes of this tutorial. Let’s
look at <em>Eubacterium</em>, <em>Porphyromonas</em>,
<em>Faecalibacteria</em>, and <em>Fusobacterium</em> for now.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chosen_genera</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Eubacterium"</span>, <span class="st">"Faecalibacterium"</span>, <span class="st">"Fusobacterium"</span>, <span class="st">"Porphyromonas"</span><span class="op">)</span></span>
<span><span class="va">mOTU_name_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">mOTU_names</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>base_name <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove</a></span><span class="op">(</span><span class="va">mOTU_names</span>, <span class="st">"unknown "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                      <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove</a></span><span class="op">(</span><span class="st">"uncultured "</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>genus_name <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/word.html" class="external-link">word</a></span><span class="op">(</span><span class="va">base_name</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">restricted_mOTU_names</span> <span class="op">&lt;-</span> <span class="va">mOTU_name_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">genus_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">chosen_genera</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span></code></pre></div>
<p>Again, while we would generally fit a model using all of our samples,
for this tutorial we are only going to consider data from a case-control
study from China.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_study_obs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Country</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHI"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Next, we want to confirm that all samples have at least one non-zero
count across the categories we’ve chosen and that all categories have at
least one non-zero count across the samples we’ve chosen.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">small_Y</span> <span class="op">&lt;-</span> <span class="va">wirbel_otu</span><span class="op">[</span><span class="va">ch_study_obs</span>, <span class="va">restricted_mOTU_names</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">small_Y</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># no samples have a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">small_Y</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># one category has a count sum of 0</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span></span>
<span><span class="va">category_to_rm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">small_Y</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">small_Y</span> <span class="op">&lt;-</span> <span class="va">small_Y</span><span class="op">[</span>, <span class="op">-</span><span class="va">category_to_rm</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">small_Y</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<p>The function that we use to fit our model is called
<code>emuFit</code>. It can accept your data in various forms, and here
we will show how to use it with data frames as input. Check out the
<code>phyloseq</code> vignette if you’d like to know how
<code>radEmu</code> plays with <code>phyloseq</code> objects! One
version of inputs to <code>emuFit</code> are</p>
<ul>
<li>
<code>formula</code>: This is a formula telling radEmu what
predictors to use in its model. We are using Group, which is an
indicator for case (CRC) vs control (CTR).</li>
<li>
<code>data</code>: A dataframe containing information on our
predictors. Recall that here we’re only looking observations from the
Chinese study.</li>
<li>
<code>Y</code>: A matrix or dataframe containing our observed
abundance data (e.g., counts or depth measurements). The rows give the
observations (samples), and the columns give the categories
(taxa/mOTUs). Here we are only considering the observations from the
Chinese study and the <em>Eubacterium</em>, <em>Porphyromonas</em>,
<em>Faecalibacteria</em>, and <em>Fusobacterium</em> genera. Note that
<code>Y</code> doesn’t have to be integer-valued (counts)!</li>
</ul>
<p>and some optional arguments include</p>
<ul>
<li>
<code>run_score_tests</code>: A logical value denoting whether or
not to run score tests. Score tests are awesome in their error rate
control (including with small sample sizes; though of course larger
sample sizes always give better power), but require refitting the model,
so can require some compute time.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span>, </span>
<span>                 data <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">[</span><span class="va">ch_study_obs</span>, <span class="op">]</span>,</span>
<span>                 Y <span class="op">=</span> <span class="va">small_Y</span>,</span>
<span>                 run_score_tests <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></code></pre></div>
<p>Let’s check out what this object looks like!</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_fit</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; emuFit(Y = small_Y, formula = ~Group, data = wirbel_sample[ch_study_obs, </span></span>
<span><span class="co">#&gt;     ], run_score_tests = FALSE)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficient estimates with the largest magnitudes:</span></span>
<span><span class="co">#&gt;    covariate                                                category</span></span>
<span><span class="co">#&gt; 24  GroupCRC               unknown Porphyromonas [meta_mOTU_v2_5431]</span></span>
<span><span class="co">#&gt; 8   GroupCRC     Fusobacterium sp. oral taxon 370 [ref_mOTU_v2_1403]</span></span>
<span><span class="co">#&gt; 19  GroupCRC                 Fusobacterium varium [ref_mOTU_v2_4311]</span></span>
<span><span class="co">#&gt; 12  GroupCRC                Porphyromonas somerae [ref_mOTU_v2_2101]</span></span>
<span><span class="co">#&gt; 9   GroupCRC         Fusobacterium gonidiaformans [ref_mOTU_v2_1404]</span></span>
<span><span class="co">#&gt; 13  GroupCRC                Porphyromonas uenonis [ref_mOTU_v2_2102]</span></span>
<span><span class="co">#&gt; 3   GroupCRC Fusobacterium nucleatum s. nucleatum [ref_mOTU_v2_0777]</span></span>
<span><span class="co">#&gt; 31  GroupCRC               unknown Porphyromonas [meta_mOTU_v2_6490]</span></span>
<span><span class="co">#&gt; 43  GroupCRC               unknown Porphyromonas [meta_mOTU_v2_7656]</span></span>
<span><span class="co">#&gt; 2   GroupCRC  Fusobacterium nucleatum s. animalis [ref_mOTU_v2_0776]</span></span>
<span><span class="co">#&gt; 28  GroupCRC            unknown Faecalibacterium [meta_mOTU_v2_5815]</span></span>
<span><span class="co">#&gt; 27  GroupCRC            unknown Faecalibacterium [meta_mOTU_v2_5779]</span></span>
<span><span class="co">#&gt; 46  GroupCRC            unknown Faecalibacterium [meta_mOTU_v2_7718]</span></span>
<span><span class="co">#&gt; 36  GroupCRC                 unknown Eubacterium [meta_mOTU_v2_7116]</span></span>
<span><span class="co">#&gt; 1   GroupCRC Fusobacterium nucleatum s. vincentii [ref_mOTU_v2_0754]</span></span>
<span><span class="co">#&gt; 25  GroupCRC                 unknown Eubacterium [meta_mOTU_v2_5463]</span></span>
<span><span class="co">#&gt; 6   GroupCRC                      Eubacterium sp. [ref_mOTU_v2_1395]</span></span>
<span><span class="co">#&gt; 21  GroupCRC                Porphyromonas uenonis [ref_mOTU_v2_4616]</span></span>
<span><span class="co">#&gt; 35  GroupCRC             Eubacterium sp. CAG:581 [meta_mOTU_v2_7088]</span></span>
<span><span class="co">#&gt; 15  GroupCRC               Eubacterium ventriosum [ref_mOTU_v2_4204]</span></span>
<span><span class="co">#&gt;    category_num   estimate        se       lower       upper</span></span>
<span><span class="co">#&gt; 24           24  6.2298815 0.9896555  4.29019247  8.16957058</span></span>
<span><span class="co">#&gt; 8             8  5.2672067 0.8352017  3.63024151  6.90417195</span></span>
<span><span class="co">#&gt; 19           19  4.5652857 0.9714733  2.66123301  6.46933841</span></span>
<span><span class="co">#&gt; 12           12  4.3176588 0.8683653  2.61569416  6.01962339</span></span>
<span><span class="co">#&gt; 9             9  3.5175395 0.8030406  1.94360888  5.09147007</span></span>
<span><span class="co">#&gt; 13           13  3.4508481 0.7990141  1.88480931  5.01688688</span></span>
<span><span class="co">#&gt; 3             3  3.0864020 1.0274037  1.07272775  5.10007627</span></span>
<span><span class="co">#&gt; 31           31  3.0279912 0.6352260  1.78297115  4.27301134</span></span>
<span><span class="co">#&gt; 43           43  2.7082209 0.9140368  0.91674164  4.49970009</span></span>
<span><span class="co">#&gt; 2             2  2.3131683 0.8270260  0.69222720  3.93410949</span></span>
<span><span class="co">#&gt; 28           28 -2.1162233 0.6522106 -3.39453255 -0.83791411</span></span>
<span><span class="co">#&gt; 27           27 -2.0205819 0.5738727 -3.14535163 -0.89581214</span></span>
<span><span class="co">#&gt; 46           46 -1.8206852 0.6495036 -3.09368882 -0.54768155</span></span>
<span><span class="co">#&gt; 36           36  1.6929902 0.8145007  0.09659813  3.28938223</span></span>
<span><span class="co">#&gt; 1             1  1.4863759 0.8965269 -0.27078456  3.24353643</span></span>
<span><span class="co">#&gt; 25           25 -1.2363804 0.6871052 -2.58308180  0.11032097</span></span>
<span><span class="co">#&gt; 6             6  1.1295208 0.9562351 -0.74466558  3.00370723</span></span>
<span><span class="co">#&gt; 21           21  0.9864200 0.8704039 -0.71954033  2.69238029</span></span>
<span><span class="co">#&gt; 35           35 -0.9817362 1.0405201 -3.02111805  1.05764560</span></span>
<span><span class="co">#&gt; 15           15 -0.8986601 0.4381018 -1.75732384 -0.03999633</span></span>
<span><span class="co">#&gt; To obtain the entire coefficient table, use the command `emuFit_object$coef`.</span></span></code></pre></div>
<p>Now, we can easily visualize our results using the <code>plot</code>
function! So that we only produce the plot, and not the dataframe used
to produce the plots, we will explicitly extract the plots using the
<code>$</code> operator to get the <code>plots</code> component.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ch_fit</span><span class="op">)</span><span class="op">$</span><span class="va">plots</span></span>
<span><span class="co">#&gt; $p1</span></span></code></pre></div>
<p><img src="intro_radEmu_files/figure-html/unnamed-chunk-15-1.png" width="576"></p>
<p>In the plot above, it is a bit difficult to read the taxa names on
the y-axis. We can create a data frame that maps the full taxa names in
our data to simplified labels and plot using those instead, as shown
below.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#create data frame that has simplified names for taxa</span></span>
<span><span class="va">taxa_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">"category"</span> <span class="op">=</span> <span class="va">ch_fit</span><span class="op">$</span><span class="va">coef</span><span class="op">$</span><span class="va">category</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cat_small <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"mOTU_"</span>, </span>
<span>                            <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="va">category</span>, <span class="st">'mOTU_v2_'</span>, simplify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>, </span>
<span>                            <span class="st">"\\]"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">#produce plot with cleaner taxa labels</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ch_fit</span>, taxon_names <span class="op">=</span> <span class="va">taxa_names</span><span class="op">)</span><span class="op">$</span><span class="va">plots</span></span>
<span><span class="co">#&gt; $p1</span></span></code></pre></div>
<p><img src="intro_radEmu_files/figure-html/unnamed-chunk-16-1.png" width="576"></p>
<p>Interestingly, we estimate a meta-mOTU “unknown Eubacterium
[meta_mOTU_v2_7116]” assigned to Eubacteria to have a much higher ratio
of abundance (comparing CRC group to control) than is typical across the
mOTUs we included in this analysis.</p>
<p>The confidence interval for this effect does not include zero – but
(!!!) the kind of confidence interval that is returned by default by
emuFit is not extremely reliable when counts are very skewed or sample
size is small-to-moderate.</p>
<p>To investigate further, let’s run a robust score test, which is more
reliable in these settings (but also takes more time because apparently
we can’t have nice things). For comparison, we’ll also test the mOTU
“Fusobacterium nucleatum s. nucleatum [ref_mOTU_v2_0777]”, which we also
estimate to have a much larger ratio of concentrations across groups
than is typical among the taxa we included in this model fit.</p>
<p>To set up this test, we can again run <code>emuFit</code>, giving it
the fitted values that it’s already found:</p>
<ul>
<li>
<code>formula</code>, <code>data</code> and <code>Y</code> are as
before</li>
<li>
<code>B</code> is our previous fitted object (the output of
<code>emuFit</code>)</li>
<li>
<code>test_kj</code> a dataframe listing the indices of the
parameters (in <code>ch_fit$B</code>) that we want to test. Below we
show how to identify these, but <code>j = 3</code> is F. nucleatum and
<code>j = 36</code> is the <em>Eubacterium</em> meta mOTU. Note that
while <code>test_kj</code> was an optional argument in previous software
versions, in <code>radEmu</code> v2.0.0.0 and forward
<code>test_kj</code> is required when
<code>run_score_tests = TRUE</code>.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">taxa_to_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">restricted_mOTU_names</span>, <span class="st">"0777"</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                  <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">restricted_mOTU_names</span>, <span class="st">"7116"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_design_matrix.html">make_design_matrix</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span>, data <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">[</span><span class="va">ch_study_obs</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "(Intercept)" "GroupCRC"</span></span>
<span><span class="va">covariate_to_test</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="co"># we see in the previous line that the covariate we care about corresponds to the second column of the design matrix</span></span>
<span><span class="va">ch_fit</span><span class="op">$</span><span class="va">B</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span></span>
<span><span class="co">#&gt; [1] "(Intercept)" "GroupCRC"</span></span>
<span><span class="va">two_robust_score_tests</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span>,</span>
<span>                                 data <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">[</span><span class="va">ch_study_obs</span>, <span class="op">]</span>,</span>
<span>                                 B <span class="op">=</span> <span class="va">ch_fit</span><span class="op">$</span><span class="va">B</span>,</span>
<span>                                 test_kj <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>k <span class="op">=</span> <span class="va">covariate_to_test</span>, </span>
<span>                                                      j <span class="op">=</span> <span class="va">taxa_to_test</span><span class="op">)</span>, </span>
<span>                                 Y <span class="op">=</span> <span class="va">small_Y</span><span class="op">)</span></span></code></pre></div>
<p>Let’s take a look at the test output.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">two_robust_score_tests</span><span class="op">$</span><span class="va">coef</span><span class="op">[</span><span class="va">taxa_to_test</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"covariate"</span>, <span class="st">"category"</span>, <span class="st">"estimate"</span>, <span class="st">"pval"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt;    covariate                                                category estimate</span></span>
<span><span class="co">#&gt; 3   GroupCRC Fusobacterium nucleatum s. nucleatum [ref_mOTU_v2_0777] 3.086402</span></span>
<span><span class="co">#&gt; 36  GroupCRC                 unknown Eubacterium [meta_mOTU_v2_7116] 1.692991</span></span>
<span><span class="co">#&gt;          pval</span></span>
<span><span class="co">#&gt; 3  0.07423599</span></span>
<span><span class="co">#&gt; 36 0.30168308</span></span></code></pre></div>
<p>The <em>Fusobacterium nucleatum</em> mOTU has a robust score test
p-value of 0.07, while the unknown Eubacterium mOTU has a robust score
test p-value of 0.3. Does this make sense? Let’s investigate further by
looking at the Eubacterium mOTU counts by Group.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">wirbel_otu</span><span class="op">[</span><span class="va">ch_study_obs</span>, <span class="st">"unknown Eubacterium [meta_mOTU_v2_7116]"</span><span class="op">]</span>,</span>
<span>           group <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Group</span><span class="op">[</span><span class="va">ch_study_obs</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>eubact_present <span class="op">=</span> <span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group</span>, <span class="va">eubact_present</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   group, eubact_present [3]</span></span></span>
<span><span class="co">#&gt;   group eubact_present     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CTR   FALSE             54</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CRC   FALSE             71</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CRC   TRUE               1</span></span></code></pre></div>
<p>We only detect this meta-mOTU in a single sample in the Chinese study
cohort! So, yes – it makes sense that our test returns a relatively
large p-value. Good job, <code>emuFit</code>!</p>
<p>Now let’s look at <em>F. nucleatum</em>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">wirbel_otu</span><span class="op">[</span><span class="va">ch_study_obs</span>, </span>
<span>                               <span class="st">"Fusobacterium nucleatum s. nucleatum [ref_mOTU_v2_0777]"</span><span class="op">]</span>,</span>
<span>           group <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Group</span><span class="op">[</span><span class="va">ch_study_obs</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>fuso_present <span class="op">=</span> <span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group</span>, <span class="va">fuso_present</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   group, fuso_present [4]</span></span></span>
<span><span class="co">#&gt;   group fuso_present     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CTR   FALSE           53</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CTR   TRUE             1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CRC   FALSE           59</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> CRC   TRUE            13</span></span></code></pre></div>
<p>This also makes sense given what we found – <em>F. nucleatum</em>
shows up in a sizeable minority of CRC cases, whereas Wirbel et al
detect it in only one control participant.</p>
<p>We could run robust score tests for every taxon in this analysis, but
it will take a longer amount of time to run. The code below will not run
in the vignette, but feel free to run it on your own.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span>, </span>
<span>                   data <span class="op">=</span> <span class="va">wirbel_sample</span><span class="op">[</span><span class="va">ch_study_obs</span>, <span class="op">]</span>,</span>
<span>                   B <span class="op">=</span> <span class="va">ch_fit</span>,</span>
<span>                   Y <span class="op">=</span> <span class="va">small_Y</span>,</span>
<span>                   run_score_tests <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="a-more-reasonable-model">A more reasonable model<a class="anchor" aria-label="anchor" href="#a-more-reasonable-model"></a>
</h2>
<p>In the above analysis, we provided a basic illustration of our method
looking at a small number of taxa, in a subset of samples (one study out
of five). However, if we we’re truly interested in identifying taxa that
are unusually abundant in either CRC cases or controls, it would make
sense to compare across study populations that are similar in their sex,
age, BMI, country and whether the samples were provided before
undergoing colonoscopy or after. That’s what we did in our manuscript!
Code to fit this model is below.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span> <span class="op">+</span> <span class="va">Study</span> <span class="op">+</span> <span class="va">Gender</span> <span class="op">+</span></span>
<span>                    <span class="va">Age_spline.1</span> <span class="op">+</span> <span class="va">Age_spline.2</span> <span class="op">+</span> </span>
<span>                    <span class="va">BMI_spline.1</span> <span class="op">+</span> <span class="va">BMI_spline.2</span> <span class="op">+</span> <span class="va">Sampling</span>,</span>
<span>                  data <span class="op">=</span> <span class="va">wirbel_sample</span>, </span>
<span>                  Y <span class="op">=</span> <span class="va">wirbel_otu</span><span class="op">[</span>, <span class="va">restricted_mOTU_names</span><span class="op">]</span>,</span>
<span>                  run_score_tests <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Note that we wanted to allow in non-linear trends in age and BMI,
which we did using B-splines. If you’re interested in doing something
similar, you could adapt the code below.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age_spline</span> <span class="op">&lt;-</span> <span class="fu">splines2</span><span class="fu">::</span><span class="fu">bSpline</span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Age</span>, degree <span class="op">=</span> <span class="fl">1</span>, knots <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">Age</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">age_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bmi_spline</span> <span class="op">&lt;-</span> <span class="fu">splines2</span><span class="fu">::</span><span class="fu">bSpline</span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">BMI</span>, degree <span class="op">=</span> <span class="fl">1</span>, knots <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">$</span><span class="va">BMI</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">bmi_spline</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Clausen, Amy D Willis, Sarah Teichman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
