<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to radEmu with phyloseq • radEmu</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to radEmu with phyloseq">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">radEmu</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/intro_radEmu_with_phyloseq.html">Introduction to radEmu with phyloseq</a></li>
    <li><a class="dropdown-item" href="../articles/intro_radEmu.html">Introduction to radEmu</a></li>
    <li><a class="dropdown-item" href="../articles/parallel_radEmu.html">Parallelizing computation for score tests with radEmu</a></li>
    <li><a class="dropdown-item" href="../articles/radEmu_clustered_data.html">Using radEmu on clustered data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/statdivlab/radEmu/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to radEmu with phyloseq</h1>
                        <h4 data-toc-skip class="author">David Clausen,
Sarah Teichman and Amy Willis</h4>
            
            <h4 data-toc-skip class="date">2024-10-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statdivlab/radEmu/blob/main/vignettes/intro_radEmu_with_phyloseq.Rmd" class="external-link"><code>vignettes/intro_radEmu_with_phyloseq.Rmd</code></a></small>
      <div class="d-none name"><code>intro_radEmu_with_phyloseq.Rmd</code></div>
    </div>

    
    
<p>First, we will install <code>radEmu</code>, if we haven’t
already.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!require("remotes", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("remotes")</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># remotes::install_github("statdivlab/radEmu")</span></span></code></pre></div>
<p>Next, we can load <code>radEmu</code> as well as the
<code>tidyverse</code> package suite.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statdivlab/radEmu" class="external-link">radEmu</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette provides an introduction to using <code>radEmu</code>
for differential abundance analysis using a <code>phyloseq</code> data
object. For more in-depth explanations of how this software works and
details on this analysis, see the vignette “intro_radEmu.Rmd”.</p>
<p>In this lab we’ll explore a <a href="https://www.nature.com/articles/s41591-019-0406-6" class="external-link">dataset
published by Wirbel et al. (2019)</a>. This is a meta-analysis of
case-control studies, meaning that Wirbel et al. collected raw
sequencing data from studies other researchers conducted and re-analyzed
it (in this case, they also collected some new data of their own).</p>
<p>Wirbel et al. published two pieces of data we’ll focus on today:</p>
<ul>
<li>metadata giving demographics and other information about
participants</li>
<li>a mOTU (metagenomic OTU) table</li>
</ul>
<p>In the manuscript, we looked at differential abundance across
otherwise similar colorectal cancer and non-cancer control study
participants for the 849 mOTUs that Wirbel et al. published. For the
purpose of having a streamlined tutorial, we will only look at a subset
of those 849 mOTUs in this vignette.</p>
</div>
<div class="section level2">
<h2 id="loading-and-exploring-data">Loading and exploring data<a class="anchor" aria-label="anchor" href="#loading-and-exploring-data"></a>
</h2>
<p>Note that in order to follow along with this tutorial (but not to use
<code>radEmu</code>!) you will need to have <code>phyloseq</code>
installed. We will check if you have <code>phyloseq</code> installed,
and if you do not then you can read the following code but it will not
be run.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">phy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"phyloseq"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">==</span> <span class="cn">TRUE</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [1] "phyloseq is installed: TRUE"</span></span></code></pre>
<p>Now that we have loaded the <code>phyloseq</code> package, we will
create our <code>phyloseq</code> data object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wirbel_otu</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wirbel_taxonomy</span><span class="op">)</span></span>
<span><span class="va">wirbel_phylo</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html" class="external-link">phyloseq</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">wirbel_sample</span><span class="op">)</span>,</span>
<span>                                   <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_otu</span>, taxa_are_rows <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>                                   <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">wirbel_taxonomy</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">wirbel_phylo</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 845 taxa and 566 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 566 samples by 14 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 845 taxa by 7 taxonomic ranks ]</span></span></code></pre></div>
<p>We’ll start by looking at the metadata.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 566  14</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             Sample_ID External_ID Age Gender BMI Country  Study</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0 CCIS00146684ST-4-0      FR-726  72      F  25     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0 CCIS00281083ST-3-0      FR-060  53      M  32     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0 CCIS02124300ST-4-0      FR-568  35      M  23     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0 CCIS02379307ST-4-0      FR-828  67      M  28     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0 CCIS02856720ST-4-0      FR-027  74      M  27     FRA FR-CRC</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0 CCIS03473770ST-4-0      FR-192  29      M  24     FRA FR-CRC</span></span>
<span><span class="co">#&gt;                    Group Library_Size Age_spline.1 Age_spline.2 BMI_spline.1</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0   CTR     35443944  -0.19755428    0.7389621   1.18982420</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0   CTR     19307896  -0.08126128   -0.6818534  -1.40679307</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0   CTR     42141246  -2.17453529   -0.6818534   0.45476676</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0   CRC      4829533   0.67464323   -0.1490476   0.07698823</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0   CTR     34294675  -0.54643328    1.0941660   0.44793355</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0   CTR     20262319  -2.87229329   -0.6818534   0.95261443</span></span>
<span><span class="co">#&gt;                    BMI_spline.2 Sampling</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0   -0.5606919   BEFORE</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0    2.0039136   BEFORE</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0   -0.6706035   BEFORE</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0    0.5384247   BEFORE</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0    0.1720525   BEFORE</span></span>
<span><span class="co">#&gt; CCIS03473770ST.4.0   -0.6706035   BEFORE</span></span></code></pre></div>
<p>We can see that this dataset includes
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>566</mn><annotation encoding="application/x-tex">566</annotation></semantics></math>
observations and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>14</mn><annotation encoding="application/x-tex">14</annotation></semantics></math>
variables.</p>
<p>Now let’s load the mOTU table.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 566 845</span></span>
<span><span class="co"># let's check out a subset</span></span>
<span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt; OTU Table:          [3 taxa and 5 samples]</span></span>
<span><span class="co">#&gt;                      taxa are columns</span></span>
<span><span class="co">#&gt;                    Streptococcus anginosus [ref_mOTU_v2_0004]</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0                                          0</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0                                          0</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0                                          2</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0                                          0</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0                                          1</span></span>
<span><span class="co">#&gt;                    Enterobacteriaceae sp. [ref_mOTU_v2_0036]</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0                                         3</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0                                         0</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0                                         5</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0                                         5</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0                                       675</span></span>
<span><span class="co">#&gt;                    Citrobacter sp. [ref_mOTU_v2_0076]</span></span>
<span><span class="co">#&gt; CCIS00146684ST.4.0                                  0</span></span>
<span><span class="co">#&gt; CCIS00281083ST.3.0                                  0</span></span>
<span><span class="co">#&gt; CCIS02124300ST.4.0                                  0</span></span>
<span><span class="co">#&gt; CCIS02379307ST.4.0                                  0</span></span>
<span><span class="co">#&gt; CCIS02856720ST.4.0                                  0</span></span></code></pre></div>
<p>We can see that this table has
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>566</mn><annotation encoding="application/x-tex">566</annotation></semantics></math>
samples (just like the metadata) and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>845</mn><annotation encoding="application/x-tex">845</annotation></semantics></math>
mOTUs. Let’s save these mOTU names in a vector.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mOTU_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Finally, we can check out the taxonomy table.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Taxonomy Table:     [6 taxa by 7 taxonomic ranks]:</span></span>
<span><span class="co">#&gt;                                                     domain     phylum          </span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          "Bacteria" "Bacillota"     </span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           "Bacteria" "Pseudomonadota"</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  "Bacteria" "Pseudomonadota"</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] "Bacteria" "Pseudomonadota"</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]            "Bacteria" "Bacillota"     </span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         "Bacteria" "Bacillota"     </span></span>
<span><span class="co">#&gt;                                                     class                </span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          "Bacilli"            </span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           "Gammaproteobacteria"</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  "Gammaproteobacteria"</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] "Gammaproteobacteria"</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]            "Bacilli"            </span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         "Bacilli"            </span></span>
<span><span class="co">#&gt;                                                     order             </span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          "Lactobacillales" </span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           "Enterobacterales"</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  "Enterobacterales"</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] "Enterobacterales"</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]            "Lactobacillales" </span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         "Lactobacillales" </span></span>
<span><span class="co">#&gt;                                                     family              </span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          "Streptococcaceae"  </span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           "Enterobacteriaceae"</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  "Enterobacteriaceae"</span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] "Enterobacteriaceae"</span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]            "Enterococcaceae"   </span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         "Lactobacillaceae"  </span></span>
<span><span class="co">#&gt;                                                     genus               </span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          "Streptococcus"     </span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           "Enterobacteriaceae"</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  "Citrobacter"       </span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] "Klebsiella"        </span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]            "Enterococcus"      </span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         "Lactobacillus"     </span></span>
<span><span class="co">#&gt;                                                     species                       </span></span>
<span><span class="co">#&gt; Streptococcus anginosus [ref_mOTU_v2_0004]          "Streptococcus anginosus"     </span></span>
<span><span class="co">#&gt; Enterobacteriaceae sp. [ref_mOTU_v2_0036]           "Enterobacteriaceae bacterium"</span></span>
<span><span class="co">#&gt; Citrobacter sp. [ref_mOTU_v2_0076]                  "Citrobacter sp."             </span></span>
<span><span class="co">#&gt; Klebsiella michiganensis/oxytoca [ref_mOTU_v2_0079] "Klebsiella michiganensis"    </span></span>
<span><span class="co">#&gt; Enterococcus faecalis [ref_mOTU_v2_0116]            "Enterococcus faecalis"       </span></span>
<span><span class="co">#&gt; Lactobacillus salivarius [ref_mOTU_v2_0125]         "Ligilactobacillus salivarius"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fitting-a-model">Fitting a model<a class="anchor" aria-label="anchor" href="#fitting-a-model"></a>
</h2>
<p><code>radEmu</code> is a package that can be used to estimate
fold-differences in the abundance of microbial taxa between levels of a
covariate. In this analysis, the covariate that we are primarily
interested in is whether a sample is from a case of colorectal cancer or
a control. We will make control (“CTR”) the reference category:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">wirbel_phylo</span><span class="op">)</span><span class="op">$</span><span class="va">Group</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CTR"</span>,<span class="st">"CRC"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>While in general we would fit a model to all mOTUs, we are going to
subset to some specific genera for the purposes of this tutorial. Let’s
look at <em>Eubacterium</em>, <em>Porphyromonas</em>,
<em>Faecalibacteria</em>, and <em>Fusobacterium</em> for now.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chosen_genera</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Eubacterium"</span>, <span class="st">"Faecalibacterium"</span>, <span class="st">"Fusobacterium"</span>, <span class="st">"Porphyromonas"</span><span class="op">)</span></span>
<span><span class="va">wirbel_restrict</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html" class="external-link">subset_taxa</a></span><span class="op">(</span><span class="va">wirbel_phylo</span>, <span class="va">genus</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">chosen_genera</span><span class="op">)</span></span></code></pre></div>
<p>Again, while we would generally fit a model using all of our samples,
for this tutorial we are only going to consider data from a case-control
study from China.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wirbel_china</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html" class="external-link">subset_samples</a></span><span class="op">(</span><span class="va">wirbel_restrict</span>, <span class="va">Country</span> <span class="op">==</span> <span class="st">"CHI"</span><span class="op">)</span></span></code></pre></div>
<p>Next, we want to confirm that all samples have at least one non-zero
count across the categories we’ve chosen and that all categories have at
least one non-zero count across the samples we’ve chosen.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_china</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># no samples have a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_china</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># one category has a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="va">category_to_rm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_china</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">wirbel_china</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html" class="external-link">subset_taxa</a></span><span class="op">(</span><span class="va">wirbel_china</span>, <span class="va">species</span> <span class="op">!=</span> <span class="va">category_to_rm</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">wirbel_china</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="co"># now no categories have a count sum of 0 </span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<p>The function that we use to fit our model is called
<code>emuFit</code>. It can accept your data in various forms, and here
we will show how to use it with a <code>phyloseq</code> object as
input.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/emuFit.html">emuFit</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="op">~</span> <span class="va">Group</span>, </span>
<span>                 Y <span class="op">=</span> <span class="va">wirbel_china</span>, </span>
<span>                 run_score_tests <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></code></pre></div>
<p>The way to access estimated coefficients and confidence intervals
from the model is with <code>ch_fit$coef</code>. Let’s plot our
results:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ch_df</span> <span class="op">&lt;-</span> <span class="va">ch_fit</span><span class="op">$</span><span class="va">coef</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Genus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">wirbel_china</span><span class="op">)</span><span class="op">[</span>, <span class="fl">6</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="co"># add genus name to output from emuFit</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cat_small <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"mOTU_"</span>, </span>
<span>                            <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="va">category</span>, <span class="st">'mOTU_v2_'</span>, simplify <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>, </span>
<span>                            <span class="st">"\\]"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cat_small <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cat_small</span>, levels <span class="op">=</span> <span class="va">cat_small</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">Genus</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="co"># reorder mOTU categories by genus</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">ch_df</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">cat_small</span>, y <span class="op">=</span> <span class="va">estimate</span>, color <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_linerange.html" class="external-link">geom_errorbar</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">cat_small</span>, ymin <span class="op">=</span> <span class="va">lower</span>, ymax <span class="op">=</span> <span class="va">upper</span>, color <span class="op">=</span> <span class="va">Genus</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Category"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Estimate"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="intro_radEmu_with_phyloseq_files/figure-html/unnamed-chunk-15-1.png" width="672"></p>
<p>If you’d like to see more explanations of the <code>radEmu</code>
software and additional analyses of this data, check out the vignette
“intro_radEmu.Rmd”.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Clausen, Amy Willis, Sarah Teichman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
